tests/core/full_node/test_mempool.py

```diff
@@ -1878,3 +1932,198 @@ def test_parse_unknown_condition(self):
 
             with pytest.raises(ValidationError):
                 cost, args = parse_condition_args(SExp.to([]), opcode, False)
+
+
+# the following tests generate generator programs and run them through get_name_puzzle_conditions()
+
+COST_PER_BYTE = 12000
+MAX_BLOCK_COST_CLVM = 11000000000
+
+
+def generator_condition_tester(
+    conditions: str, safe_mode: bool = False, quote: bool = True, max_cost: int = MAX_BLOCK_COST_CLVM
+) -> NPCResult:
+    prg = f"(q ((0x0101010101010101010101010101010101010101010101010101010101010101 {'(q ' if quote else ''} {conditions} {')' if quote else ''} 123 (() (q . ())))))"  # noqa
+    print(f"program: {prg}")
+    program = SerializedProgram.from_bytes(binutils.assemble(prg).as_bin())
+    generator = BlockGenerator(program, [])
+    print(f"len: {len(bytes(program))}")
+    npc_result: NPCResult = get_name_puzzle_conditions(
+        generator, max_cost, cost_per_byte=COST_PER_BYTE, safe_mode=safe_mode
+    )
+    return npc_result
+
+
+class TestGeneratorConditions:
+    def test_duplicate_height_time_conditions(self):
+        # ASSERT_SECONDS_RELATIVE
```

Comment:
Maybe we need to define these values in some python header file so we don't have to write `80, 81, 82, 83`. They exist as an `Enum`, but using them is awkward... a simpler version might be in order (even if slightly redundant). It might contain

`ASSERT_SECONDS_ABSOLUTE = ConditionOpcode.ASSERT_SECONDS_ABSOLUTE.value`

etc.

Reply:
yeah. I specifically left the comments here to make them easy to find. ideally we would have constants. However, in some (future) tests, they also appear in strings. I suppose they could turn into f-strings with `{constant}` instead.